celery/backends/azureblockblob.py

```diff
@@ -0,0 +1,160 @@
+"""The Azure Storage Block Blob backend for Celery."""
+from __future__ import absolute_import, unicode_literals
+
+import re
+
+from kombu.utils import cached_property
+from kombu.utils.encoding import bytes_to_str
+
+from celery.exceptions import ImproperlyConfigured
+from celery.utils.log import get_logger
+
+from .base import KeyValueStoreBackend
+
+try:
+    import azure.storage as azurestorage
+    from azure.common import AzureMissingResourceHttpError
+    from azure.storage.blob import BlockBlobService
+    from azure.storage.common.retry import ExponentialRetry
+except ImportError:  # pragma: no cover
+    azurestorage = BlockBlobService = ExponentialRetry = \
+        AzureMissingResourceHttpError = None  # noqa
+
+__all__ = ("AzureBlockBlobBackend",)
+
+
+CONF_CONTAINER_NAME = "azureblockblob_container_name"
+CONF_RETRY_INITIAL_BACKOFF_SEC = "azureblockblob_retry_initial_backoff_sec"
+CONF_RETRY_INCREMENT_BASE = "azureblockblob_retry_increment_base"
+CONF_RETRY_MAX_ATTEMPTS = "azureblockblob_retry_max_attempts"
+
+DEFAULT_CONTAINER_NAME = "celery"
+DEFAULT_RETRY_INITIAL_BACKOFF_SEC = 2
+DEFAULT_RETRY_INCREMENT_BASE = 2
+DEFAULT_RETRY_MAX_ATTEMPTS = 3
+
+URL_RE = re.compile(r"azureblockblob://(?P<connection_string>.+)")
+
+LOGGER = get_logger(__name__)
+
+
+class AzureBlockBlobBackend(KeyValueStoreBackend):
+    """Azure Storage Block Blob backend for Celery."""
+
+    def __init__(self,
+                 url=None,
+                 container_name=None,
+                 retry_initial_backoff_sec=None,
+                 retry_increment_base=None,
+                 retry_max_attempts=None,
+                 *args,
+                 **kwargs):
+        super(AzureBlockBlobBackend, self).__init__(*args, **kwargs)
+
+        if azurestorage is None:
+            raise ImproperlyConfigured(
+                "You need to install the azure-storage library to use the "
+                "AzureBlockBlob backend")
+
+        conf = self.app.conf
+
+        self._connection_string = self._parse_url(url)
+
+        self._container_name = (
+            container_name or
+            conf.get(CONF_CONTAINER_NAME) or
```

Comment:
It's slightly simpler to just use the default value of the `get` method here:

```
conf.get(CONF_CONTAINER_NAME, DEFAULT_CONTAINER_NAME)
```

Reply:
That was my initial implementation but it failed the tests since the conf value were explicitly set to None in one of the mappings that the conf access checks. Moving the default values to `app/defaults.py` fixed this. Implemented in 44517e3.